centreon/www/class/centreon-clapi/centreonLDAP.class.php

```diff
@@ -1,164 +1,140 @@
-        $this->baseParams = array(
-            'alias' => '',
-            'bind_dn' => '',
-            'bind_pass' => '',
-            'group_base_search' => '',
-            'group_filter' => '',
-            'group_member' => '',
-            'group_name' => '',
-            'ldap_auto_import' => '',
-            'ldap_contact_tmpl' => '',
-            'ldap_default_cg' => '',
-            'ldap_dns_use_domain' => '',
-            'ldap_connection_timeout' => '',
-            'ldap_search_limit' => '',
-            'ldap_search_timeout' => '',
-            'ldap_srv_dns' => '',
-            'ldap_store_password' => '',
-            'ldap_template' => '',
-            'protocol_version' => '',
-            'user_base_search' => '',
-            'user_email' => '',
-            'user_filter' => '',
-            'user_firstname' => '',
-            'user_lastname' => '',
-            'user_name' => '',
-            'user_pager' => '',
-            'user_group' => ''
-        );
-        $this->serverParams = array('host_address', 'host_port', 'host_order', 'use_ssl', 'use_tls');
-        $this->action = "LDAP";
+        $this->action = 'LDAP';
     }
 
     /**
-     * Checks if configuration name is unique
+     * Get Ldap Configuration Id.
      *
      * @param string $name
-     * @param int $arId
-     * @return boolean
-     */
-    protected function isUnique($name = "", $arId = 0)
-    {
-        $stmt = $this->db->query(
-            "SELECT ar_name FROM auth_ressource WHERE ar_name = ? AND ar_id != ?",
-            array($name, $arId)
-        );
-        $res = $stmt->fetchAll();
-        if (count($res)) {
-            return false;
-        }
-        return true;
-    }
-
-    /**
-     * Get Ldap Configuration Id
      *
-     * @param string $name
-     * @return mixed | returns null if no ldap id is found
      * @throws CentreonClapiException
+     *
+     * @return mixed returns null if no ldap id is found
      */
     public function getLdapId($name)
     {
-        $res = $this->db->query("SELECT ar_id FROM auth_ressource WHERE ar_name = ?", array($name));
+        $res = $this->db->query('SELECT ar_id FROM auth_ressource WHERE ar_name = ?', [$name]);
         $row = $res->fetch();
-        if (!isset($row['ar_id'])) {
-            return null;
+        if (! isset($row['ar_id'])) {
+            return;
         }
         $ldapId = $row['ar_id'];
         unset($res);
+
         return $ldapId;
     }
 
     /**
      * @param $id
+     *
      * @return mixed
      */
     public function getLdapServers($id)
     {
-        $query = "SELECT host_address, host_port FROM auth_ressource_host  WHERE auth_ressource_id = ?";
-        $res = $this->db->query($query, array($id));
-        $row = $res->fetchAll();
-        return $row;
+        $query = 'SELECT host_address, host_port FROM auth_ressource_host  WHERE auth_ressource_id = ?';
+        $res = $this->db->query($query, [$id]);
```

Comment:
Please use a prepare + execute to avoid this legacy call to a modified query signature from PDO 🙏